planning/behavior_velocity_planner/src/scene_module/run_out/scene.cpp

```diff
@@ -152,47 +141,30 @@ void RunOutModule::visualizeDetectionArea(const PathWithLaneId & smoothed_path)
     initial_vel, target_vel, initial_acc, planning_dec, jerk_acc, jerk_dec);
 
   if (!stop_dist) {
-    return;
+    *stop_dist = 0;
   }
 
+  // create detection area polygon
   DetectionRange da_range;
-  const float obstacle_vel_mps = planner_param_.dynamic_obstacle.max_vel_kmph / 3.6;
-  da_range.interval = planner_param_.run_out.detection_distance;
-  da_range.min_longitudinal_distance = planner_param_.vehicle_param.base_to_front;
-  da_range.max_longitudinal_distance = *stop_dist + planner_param_.run_out.stop_margin;
-  da_range.min_lateral_distance = planner_param_.vehicle_param.width / 2.0;
-  da_range.max_lateral_distance =
-    obstacle_vel_mps * planner_param_.dynamic_obstacle.max_prediction_time;
+  const auto & p = planner_param_;
+  const double obstacle_vel_mps = p.dynamic_obstacle.max_vel_kmph / 3.6;
+  da_range.interval = p.run_out.detection_distance;
+  da_range.min_longitudinal_distance =
+    p.vehicle_param.base_to_front - p.detection_area.margin_behind;
```

Comment:
CreateDetectionAreaPolygon is designed to be offset by min_longitudinal_distance. Is margin_behind added to the Polygon as expected?

Reply:
min_longitudinal_distance is the offset from base_link, so the margin is as following image.
Since lateral offset depends on the `margin_behind` and ego velocity, we need to think another way to set more precise offset. But I think this is enough as the pre-filter for now.
![image](https://user-images.githubusercontent.com/11865769/177515548-4ce431d9-4d6a-4f6c-be3b-7bb8d9ffe765.png)